awscli2: init at 2.0.26

davegallant opened the pull request:
#### Motivation for this change

"AWS CLI v2 supports configuring and using credentials through AWS SSO using the new aws configure sso command."
https://aws.amazon.com/blogs/developer/aws-cli-v2-is-now-generally-available/

Since AWS CLI version 1 will be supported for a while, it may make sense to keep this a separate derivation.

https://github.com/NixOS/nixpkgs/issues/80956

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

bhipple commented on pkgs/tools/admin/awscli2/default.nix:
```diff
@@ -0,0 +1,83 @@
+      owner = "aws";
+      repo = "aws-cli";
+      rev = version;
+      sha256 = "sha256:1ysmr17gbcj6vs9ywzwgvd9caxwxgg9bnfvvkyks4fii34ji5qq8";
```
```suggestion
      hash = "sha256:1ysmr17gbcj6vs9ywzwgvd9caxwxgg9bnfvvkyks4fii34ji5qq8";
```

bhipple commented on pkgs/tools/admin/awscli2/default.nix:
```diff
@@ -0,0 +1,83 @@
+      repo = "aws-cli";
+      rev = version;
+      sha256 = "sha256:1ysmr17gbcj6vs9ywzwgvd9caxwxgg9bnfvvkyks4fii34ji5qq8";
+    };
```
Indentation is messed up here; give it a run through https://github.com/nix-community/nixpkgs-fmt to fix automatically.

davegallant replied:
Thanks for the feedback. I applied the suggestions.

bhipple reviewed (changes requested):
Thanks for taking this on!

bhipple commented:
LGTM; can you fixup git history per https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md ?

davegallant committed:
awscli2: init at 2.0.26

Amazon released v2 of their CLI, but is still actively maintaining v1.

- https://github.com/aws/aws-cli/blob/2.0.26/CHANGELOG.rst
- https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html
- https://docs.aws.amazon.com/cli/latest/userguide/cliv2-migration.html

davegallant commented:
> LGTM; can you fixup git history per https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md ?

@bhipple Fixed :wrench:

bhipple commented:
Result of `nixpkgs-review pr 91655` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
<br>- awscli2
</details>
